File: code/geometry/dist_ps.py
def distPS(s, e, p):
	if s == e:
		return sqrt(dist2(p, s))
	se, sp = vecsub(e, s), vecsub(p, s)
	d = len2(se)
	t = min(d, max(0, dot(vecsub(p, s), vecsub(e, s))))
	return sqrt(dist2((sp[0] * d, sp[1] * d), (se[0] * t, se[1] * t))) / d

File: code/geometry_cpp/dist_ps.cpp
double distPS(ipoint s, ipoint e, ipoint p) {
	if (s == e)
		return sqrt((p - s).len2());
	auto se = e - s;
	auto sp = p - s;
	ll d = se.len2();
	ll t = min(d, max(0LL, (p - s).dot(e - s)));
	return sqrt((sp * d - se * t).len2()) / d;
}
